### PR TITLE
Stats for BigQuery results

### PR DESCRIPTION
**Pull request by SammyEK**
- [x ] fix integer type for parser
- [x]  add stats for bigquery output in INSERT INTO statements
- [ ] TODO: have the same logic in `override_result` when the destination is a BQ table

Signed-off-by: Sammy El Khammal <sammy.elkhammal@kensu.io>

### CODE REVIEW

**Review by andypetrella — APPROVED** [review]

**Review by kensustan — DISMISSED** [review]

**Comment by SammyEK** [comment]
added support of merge queries

File: kensu/google/cloud/bigquery/client.py [hunk 1]
```diff
@@ -61,12 +67,28 @@ def query(
                         kensu=kensu,
                         client=client,
                         query=query)
+
                     try:
-                        #TODO: Use sqlparse
-                        index_select = query.lower().index("select")
-                        query_without_insert = query[index_select:]
-                        logger.debug(f"Query without INSERT TO:{query_without_insert}")
+                        if query.lower().replace(" ","").replace("\n","").startswith("insertinto") :
```
Comment (andypetrella) [inline kensu/google/cloud/bigquery/client.py:72]: This is done twice I think

**Review by andypetrella — APPROVED** [review]
There is quite logic of parsing that gets embedded into a function with if elses.
Some better strategies to extract and test this should be introduce to simplify maintenance and evolutions - maybe using other parsers offering better visiting and such (cc @kensustan )

File: kensu/google/cloud/bigquery/client.py [hunk 2]
```diff
@@ -63,11 +69,22 @@ def query(
...
+                            index_select = query.lower().index("select")
+                            query_without_insert = query[index_select:]
+                            logger.debug(f"Query without INSERT TO:{query_without_insert}")
+                        elif query.lower().replace(" ","").replace("\n","").startswith("merge"):
+                            import re
+                            query_without_insert = re.findall('\(.*\)', query)[0]
```
Comment (vidma) [inline kensu/google/cloud/bigquery/client.py:78]: @SammyEK be careful that at least in generic SQL, it's possible to write a MERGE statement without the `(` and `)`, and/or the `WHEN` clauses could be INSIDE the `( ... )`
so in longer term using a proper parser would be safer here.

 see https://stackoverflow.com/questions/46613907/does-apache-spark-sql-support-merge-clause

File: kensu/google/cloud/bigquery/client.py (hunk 1 above)
Comment (vidma) [reply kensu/google/cloud/bigquery/client.py:72]: yeah could be extracted into a variable of something

**Comment by kensustan** [comment]
@andypetrella @vidma the more generic SQL parser was evoked at the cust_srv weekly meeting this morning along a more diagnostic oriented version, I suggest either another discussion with you both, Franck and Sammy or a report, as it's not a simple/binary approach, requires strategy and prioritisation

File: kensu/google/cloud/bigquery/job/bigquery_stats.py [hunk 3]
```diff
@@ -1,16 +1,18 @@
...
 import logging
 import datetime
+logger = logging.getLogger(__name__)
 
 from kensu.google.cloud.bigquery import Client
 from kensu.utils.kensu_provider import KensuProvider
 
 
-def compute_bigquery_stats(table_ref, table, client, stats_aggs, input_filters=None):
+def compute_bigquery_stats(table_ref=None, table=None, client=None, stats_aggs=None, input_filters=None, query=None):
```
Comment (kensustan) [inline kensu/google/cloud/bigquery/job/bigquery_stats.py:10]: it's good practice to inform when a value is required 
if None == query == table_ref:
    raise Exception("compute_bigquery_stats requires either a table_ref or a query")

**Review by kensustan — APPROVED** [review]

**Pull request merged**